tests/chainerx_tests/unit_tests/routines_tests/test_connection.py

```diff
@@ -141,6 +122,66 @@ def forward_chainer(self, inputs):
         return y,
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # without bias
+    chainer.testing.product([
+        chainer.testing.from_pytest_parameterize(
+            'x_shape,w_shape,b_shape,stride,pad', [
+                ((1, 3), (5, 3), None, 1, 0),
+                ((1, 3, 4), (5, 3, 2), None, 3, 2),
+                ((2, 3, 4, 4), (2, 3, 3, 3), None, 2, (2, 0)),
+                ((2, 3, 2, 6, 3), (2, 3, 1, 3, 2), None, (1, 2, 3), (2, 0, 1)),
+            ]),
+        chainer.testing.from_pytest_parameterize(
+            'in_dtypes,out_dtype', dtype_utils.result_dtypes_two_arrays)
+    ]) +
+    # with bias
+    chainer.testing.product([
+        chainer.testing.from_pytest_parameterize(
+            'x_shape,w_shape,b_shape,stride,pad', [
+                ((1, 3), (5, 3), (5,), 1, 0),
+                ((2, 3, 4), (5, 3, 1), (5,), 1, 0),
+                ((1, 3, 4), (5, 3, 2), (5,), 3, 2),
+                ((2, 3, 4, 4), (2, 3, 3, 3), (2,), 1, 0),
+                ((1, 3, 4, 4), (2, 3, 3, 3), (2,), (1, 2), 1),
+                ((1, 3, 4, 4), (2, 3, 3, 3), (2,), 2, (2, 0)),
+                ((1, 3, 2, 6, 3), (2, 3, 1, 3, 2), (2,), 2, (2, 0, 1)),
+                ((1, 3, 2, 6, 3), (2, 3, 1, 3, 2), (2,), (1, 2, 3), (2, 0, 1)),
+            ]),
+        chainer.testing.from_pytest_parameterize(
+            'in_dtypes,out_dtype', dtype_utils.result_dtypes_three_arrays)
+    ])
+))
+@chainer.testing.parameterize_pytest('cover_all', [True, False])
+class TestConv(_ConvTestBase, op_utils.ChainerOpTest):
+    pass
+
+
+# cudnnFindConvolutionForwardAlgorithmEx tends to choose an algorithm which
+# uses TensorCore when:
+# - The sizes of the input arrays are large
+# - The input `x` is aligned with NHWC data layout
+@op_utils.op_test(['cuda:0'])
+@chainer.testing.parameterize(*(
```

Comment:
It seems all those parameters can be class attributes.

Reply:
Fixed.